script.js

```diff
@@ -1,12 +1,543 @@
+  if (showOne.name.toLowerCase() < showTwo.name.toLowerCase()) {
+    return -1;
+  }
+  if (showOne.name.toLowerCase() < showTwo.name.toLowerCase()) {
+    return 1;
+  }
+  return 0;
+}
+
+// when logic is done call in the eventListener
+// tv input selector
+function addShowSelector(showData) {
+  const rootElem = document.getElementById("root");
+  // this only works if the element does not exist
+  select = document.createElement("select");
+  select.setAttribute("id", "select-tv");
+  rootElem.appendChild(select);
+
+  showData.sort(compare).forEach((show) => {
+    //  when the function processes the forEach the array will be sorted
+    let option = document.createElement("option");
+    option.innerHTML = `${show.name}`; // loops over the tv show data
+    select.appendChild(option);
+  });
+ 
+  // loose a track of the episode list because errors occur when it tries to
+  //compare the two fields and the array
+  // it also fails with the hide and show
+  select.addEventListener("change", function (event) {
+    // An array of episode container
+    let optionId = event.target.options[event.target.selectedIndex].index;
+    let showId = showData[optionId].id;
+    // for every item duplicate we need an test to see if it exists
+
+    let styleContainer = document.getElementById("styleContainer");
+    if (styleContainer !== null) {
+      styleContainer.innerHTML = ""; // its removing all the episode containers
+    }
+    getAllEpisodesApi(showId); // this call gets the all the episodes it gets the data and calls adds all episodes , add allEpisodes creates the containers
+    //--------------------------------------------------------------
+    let tvSelector = document.querySelector("#select-tv"); // hide the tv episodes when the button is pressed
+    tvSelector.style.display = "none";
+
+    // when show title is clicked we load the episodes fpr selected show
+    let btn = document.getElementById("returnBtn");
+    btn.style.display = "";
+    let select = document.getElementById("select");
+    if (select) {
+      select.style.display = "";
+    }
+
+    // this hides the tv show when the tv show is selected the episodes appear
+    let arrayOfTvContainers = document.querySelectorAll(".infoContainer");
+    Array.from(arrayOfTvContainers).forEach((element) => {
+      // converts dom element into an array in order to use the for each
+      element.style.display = "none";
+    });
+  });
+
+  return select;
+}
+
+function addFooter() {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let footer = document.getElementById("footer");
+  if (footer === null) {
+    footer = document.createElement("footer");
+    footer.setAttribute("id", "footer");
+    footer.innerHTML = "<p>https://tvmaze.com/</p>";
+    rootElem.appendChild(footer);
+  }
+}
+
+const hideEpisode = (episode) => {
+  episode.style.display = "none";
```

Comment:
It's a good practice to keep your styles in a css file and only apply classes in your code. It's a lot easier to debug styling if everything is in one place.

Reply:
Yes I agree in hindsight I would separate the styling from the JavaScript to make the code more readable.